src/middlewared/middlewared/plugins/vm/pci.py

```diff
@@ -186,3 +189,27 @@ def passthrough_device_choices(self):
     def pptdev_choices(self):
         """Available choices for PCI passthru device"""
         return self.get_all_pci_devices_details()
+
+    @accepts(Str('gpu_pci_id', empty=False))
+    @returns(List(Str('pci_ids')))
+    def get_pci_ids_for_gpu_isolation(self, gpu_pci_id):
+        """Get PCI IDs for GPU isolation"""
+        gpu = next(
+            (gpu for gpu in get_gpus() if convert_pci_id_to_vm_pci_slot(gpu['addr']['pci_slot']) == gpu_pci_id), None
+        )
+        if not gpu:
+            raise CallError(f'GPU {gpu_pci_id} not found', errno=errno.ENOENT)
```

Comment:
If this is to be called by webui then we should raise a validation error.

Reply:
Sure, although this won't be a user input - so no form is involved. Will make relevant changes, thanks!